diff --git a/CMakeLists.txt b/CMakeLists.txt
index 0c45a13..d5e40c9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -436,7 +436,7 @@ if(X11_FOUND)
 	list(APPEND MEDIASTREAMER2_INCLUDE_DIRS ${X11_INCLUDE_DIRS})
 endif()
 if(XV_FOUND)
-	list(APPEND LINK_LIBS Xv)
+	list(APPEND LINK_LIBS Xv Xext)
 	list(APPEND MEDIASTREAMER2_INCLUDE_DIRS ${XV_INCLUDE_DIRS})
 endif()
 if(GLX_FOUND)
diff --git a/src/CMakeLists.txt b/src/CMakeLists.txt
index affbc06..19ee247 100644
--- a/src/CMakeLists.txt
+++ b/src/CMakeLists.txt
@@ -533,7 +533,7 @@ if(ENABLE_SHARED)
 		$<BUILD_INTERFACE:${PROJECT_SOURCE_DIR}/include/>
 		PRIVATE ${MEDIASTREAMER2_INCLUDE_DIRS}
 	)
-	target_link_libraries(mediastreamer PRIVATE ${LINK_LIBS})
+        target_link_libraries(mediastreamer PRIVATE ${LINK_LIBS} pthread)
 	if(bzrtp_FOUND)
 		target_link_libraries(mediastreamer PRIVATE bzrtp)
 	endif()

