diff -up ./CMakeLists.txt.orig ./CMakeLists.txt
--- ./CMakeLists.txt.orig	2017-12-31 15:58:07.074428212 +0200
+++ ./CMakeLists.txt	2017-12-31 16:04:43.716458600 +0200
@@ -457,6 +457,7 @@ if(ENABLE_DEBUG_LOGS)
 endif()
 if(ENABLE_FIXED_POINT)
 	set(MS_FIXED_POINT 1)
+	set(MS_PUBLIC_CFLAGS "-DMS_FIXED_POINT")
 endif()
 if(ENABLE_NON_FREE_CODECS)
 	set(HAVE_NON_FREE_CODECS 1)
@@ -624,6 +624,16 @@ configure_file(cmake/Mediastreamer2Confi
 	@ONLY
 )
 
+set(prefix ${CMAKE_INSTALL_PREFIX})
+set(exec_prefix ${prefix}/libexec)
+set(libdir ${prefix}/lib${LIB_SUFFIX})
+set(includedir ${prefix}/include)
+
+configure_file("${CMAKE_CURRENT_SOURCE_DIR}/mediastreamer.pc.in"
+        "${CMAKE_CURRENT_BINARY_DIR}/mediastreamer.pc"
+        @ONLY
+)
+
 set(CONFIG_PACKAGE_LOCATION "${CMAKE_INSTALL_DATADIR}/Mediastreamer2/cmake")
 install(EXPORT ${EXPORT_TARGETS_NAME}Targets
 	FILE Mediastreamer2Targets.cmake
@@ -635,4 +645,9 @@ install(FILES
 	DESTINATION ${CONFIG_PACKAGE_LOCATION}
 )
 
+install(FILES
+	"${CMAKE_CURRENT_BINARY_DIR}/mediastreamer.pc"
+	DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig"
+)
+
 add_subdirectory(build)
